File: sort/applied/two_sum_two_pointers/two_pointers.java
class Solution {
  public int[] twoSum(int[] nums, int target) {
    int startIndex = 0, endIndex = nums.length - 1;
    while(startIndex < endIndex){
      int sum = nums[startIndex] + nums[endIndex];
      if(sum == target){
          return new int[]{startIndex + 1, endIndex + 1};
      }
      else if (sum < target){
          startIndex++;
      }
      else{
          endIndex--;
      }
    }
    return new int[]{};
  }
}

File: data_structures/sorted/applied/two_sum_two_pointers/two_pointers.scala
object Solution {
  def twoSum(numbers: Array[Int], target: Int): Array[Int] = {
    @annotation.tailrec
    def recursiveTwoSum(startIndex: Int, endIndex: Int): Array[Int] = {
        val sum = numbers(startIndex) + numbers(endIndex)
        if(sum < target && endIndex - startIndex > 1) recursiveTwoSum(startIndex + 1, endIndex)
        else if(sum > target && endIndex - startIndex > 1) recursiveTwoSum(startIndex, endIndex - 1)
        else Array(startIndex + 1, endIndex + 1)
    }
    recursiveTwoSum(0, numbers.length - 1)
  }
}